tests/mocks/components.py

```diff
@@ -104,6 +104,9 @@ async def stop(self, *, extra: Optional[Dict[str, Any]] = None, timeout: Optiona
     async def is_moving(self) -> bool:
         return not self.is_stopped
 
+    async def do_command(self, command: Dict[str, Any], *, timeout: Optional[float] = None, **kwargs) -> Dict[str, Any]:
+        return {"hello": "world"}
```

Comment:
I would make all the return something like `{"command": command}`, that way you can test whether commands are being successfully sent from `service -> component` and `client -> service -> component`